lib/semian/protected_resource.rb

```diff
@@ -13,7 +13,22 @@ def initialize(resource, circuit_breaker)
     end
 
     def acquire(*args, &block)
-      @circuit_breaker.acquire { @resource.acquire(*args, &block) }
+      success = true
+      @circuit_breaker.acquire do
+        begin
+          @resource.acquire(*args, &block)
+        rescue ::Semian::TimeoutError
+          success = false
+          Semian.notify(:occupied, self)
+          raise
+        end
+      end
+    rescue ::Semian::OpenCircuitError
+      success = false
+      Semian.notify(:circuit_open, self)
+      raise
+    ensure
+      Semian.notify(:success, self) if success
```

Comment:
if `acquire` raises with a different exception than you expect here, this will still notify for success... is that intended?